REC2Plugin.py

```diff
@@ -78,10 +66,10 @@ def __showSetupDialog(self):
         if setupDialog.exec_() != QDialog.Accepted:
             return
 
-        self.iface.newProject(True)
-        # If new project was not created
-        if QgsProject.instance().isDirty():
-            return
+        # Prevents Setup if layers already exists
+        layers = self.iface.legendInterface().layers()
+        if(layers):
+            QgsMapCanvas().setDestinationCrs(QgsCoordinateReferenceSystem(2193,  QgsCoordinateReferenceSystem.EpsgCrsId));
```

Comment:
This is a no-op, you are setting the destination crs on a newly allocated `QgsMapCanvas`, not on the main one. You probably meant to set the destination crs on `self.iface.mapCanvas()`